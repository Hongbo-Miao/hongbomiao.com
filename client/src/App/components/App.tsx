import React, { lazy } from 'react';
import { BrowserRouter, Redirect, Route, Switch } from 'react-router-dom';
import HmLazyComponent from '../../shared/components/LazyComponent';
import Paths from '../../shared/utils/paths';
import styles from './App.module.css';

const HmHome = lazy(() => import('../../Home/components/Home'));
const HmFooter = lazy(() => import('./Footer'));

const App: React.FC = () => (
  <div className={styles.hmApp}>
    <BrowserRouter>
      <HmLazyComponent>
        <Switch>
          <Route exact path={Paths.appRootPath} component={HmHome} />
          <Redirect to={Paths.appRootPath} />
        </Switch>
      </HmLazyComponent>
    </BrowserRouter>

    <HmLazyComponent>
      <HmFooter />
    </HmLazyComponent>
  </div>
);

export default App;
