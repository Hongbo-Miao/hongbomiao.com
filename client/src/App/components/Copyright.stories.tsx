import { storiesOf } from '@storybook/react';
import React from 'react';

import HmCopyright from './Copyright';

storiesOf('Copyright', module).add('default', () => <HmCopyright year={1990} />);
