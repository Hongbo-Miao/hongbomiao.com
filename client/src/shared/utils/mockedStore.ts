import configureStore from 'redux-mock-store';
import ReducerMe from '../../Home/types/ReducerMe.type';
import ReducerHealth from '../../health/types/ReducerHealth.type';
import RootState from '../types/RootState.type';

const mockStore = configureStore();

const me: ReducerMe = {
  name: 'Hongbo Miao',
  bio: 'Making magic happen',
};
const health: ReducerHealth = {};

const initialState: RootState = { me, health };
const mockedStore = mockStore(initialState);

export default mockedStore;
