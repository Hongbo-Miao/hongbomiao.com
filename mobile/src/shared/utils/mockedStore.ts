import configureStore from 'redux-mock-store';
import MeState from '../../Home/types/MeState.type';
import RootState from '../types/RootState.type';

const mockStore = configureStore();

const me: MeState = {
  name: 'Hongbo Miao',
  bio: 'Making magic happen',
};

const initialState: RootState = { me };
const mockedStore = mockStore(initialState);

export default mockedStore;
