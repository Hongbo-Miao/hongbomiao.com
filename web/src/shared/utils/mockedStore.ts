import configureStore from 'redux-mock-store';
import MeState from '../../Home/types/MeState.type';
import HealthState from '../../health/types/HealthState.type';
import RootState from '../types/RootState.type';

const mockStore = configureStore();

const me: MeState = {
  name: 'Hongbo Miao',
  bio: 'Making magic happen',
};
const health: HealthState = {};

const initialState: RootState = { me, health };
const mockedStore = mockStore(initialState);

export default mockedStore;
