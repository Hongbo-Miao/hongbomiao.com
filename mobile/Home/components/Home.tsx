import { Layout, Text } from '@ui-kitten/components';
import React from 'react';
import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  hmHome: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  hmName: {
    fontFamily: 'OpenSans_700Bold',
    fontSize: 35,
    marginTop: 20,
    textTransform: 'uppercase',
  },
  hmBio: {
    fontFamily: 'OpenSans_400Regular',
    fontSize: 20,
    marginTop: 20,
  },
});

const Home: React.FC = () => (
  <Layout style={styles.hmHome}>
    <Text style={styles.hmName}>Hongbo Miao</Text>
    <Text style={styles.hmBio}>Making magic happen</Text>
  </Layout>
);

export default Home;
