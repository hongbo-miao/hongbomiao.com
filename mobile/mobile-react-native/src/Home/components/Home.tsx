import { Layout, Text } from '@ui-kitten/components';
import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  hmHome: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  hmName: {
    fontFamily: 'NeoSansProBold',
    fontSize: 35,
    marginTop: 20,
    textTransform: 'uppercase',
  },
  hmBio: {
    fontFamily: 'NeoSansProRegular',
    fontSize: 20,
    marginTop: 40,
  },
});

function Home() {
  return (
    <Layout style={styles.hmHome}>
      <Text style={styles.hmName} testID="name">
        Hongbo Miao
      </Text>
      <Text style={styles.hmBio} testID="bio">
        Making magic happen
      </Text>
    </Layout>
  );
}

export default Home;
