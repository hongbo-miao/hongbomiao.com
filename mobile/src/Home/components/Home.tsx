import { Layout, Text } from '@ui-kitten/components';
import React from 'react';
import { StyleSheet } from 'react-native';
import { connect, ConnectedProps } from 'react-redux';
import RootState from '../../shared/types/RootState.type';
import MeAction from '../actions/MeAction';
import meQuery from '../queries/meQuery';

const connector = connect(
  (state: RootState) => ({
    me: state.me,
  }),
  {
    queryMe: MeAction.queryMe,
  },
);

const styles = StyleSheet.create({
  hmHome: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  hmName: {
    fontFamily: 'NeoSansProBold',
    fontSize: 35,
    marginTop: 20,
    textTransform: 'uppercase',
  },
  hmBio: {
    fontFamily: 'NeoSansProRegular',
    fontSize: 20,
    marginTop: 40,
  },
});

type Props = ConnectedProps<typeof connector>;

const Home: React.FC<Props> = (props) => {
  const { me, queryMe } = props;

  React.useEffect(() => {
    queryMe(meQuery);
  }, [queryMe]);

  const { bio, name } = me;

  return (
    <Layout style={styles.hmHome}>
      <Text style={styles.hmName} testID="name">
        {name}
      </Text>
      <Text style={styles.hmBio} testID="bio">
        {bio}
      </Text>
    </Layout>
  );
};

export default connector(Home);
