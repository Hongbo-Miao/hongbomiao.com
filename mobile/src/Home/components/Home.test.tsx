import * as eva from '@eva-design/eva';
import { render } from '@testing-library/react-native';
import { ApplicationProvider } from '@ui-kitten/components';
import React from 'react';
import { Provider } from 'react-redux';
import mockedStore from '../../shared/utils/mockedStore';
import HmHome from './Home';

describe('<HmHome />', () => {
  test('Home', () => {
    const { getByTestId, toJSON } = render(
      <Provider store={mockedStore}>
        {/* eslint-disable-next-line react/jsx-props-no-spreading */}
        <ApplicationProvider {...eva} theme={eva.light}>
          <HmHome />
        </ApplicationProvider>
      </Provider>
    );
    expect(getByTestId('bio').props.children).toBe('Making magic happen');
    expect(toJSON()).toMatchSnapshot();
  });
});
