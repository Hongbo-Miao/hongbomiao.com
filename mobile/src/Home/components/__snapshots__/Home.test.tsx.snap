// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HmHome /> Home 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    appearance="default"
    level="1"
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
        },
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#222B45",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
          },
          Object {
            "fontFamily": "NeoSansProBold",
            "fontSize": 35,
            "marginTop": 20,
            "textTransform": "uppercase",
          },
        ]
      }
      testID="name"
    >
      Hongbo Miao
    </Text>
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#222B45",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
          },
          Object {
            "fontFamily": "NeoSansProRegular",
            "fontSize": 20,
            "marginTop": 40,
          },
        ]
      }
      testID="bio"
    >
      Making magic happen
    </Text>
  </View>
</View>
`;
